### PR TITLE
[MERGE][REF] web,*: user service is no longer a service

### PR DESCRIPTION
**Pull request by aab-odoo**
This PR aims at simplifying the way developpers can access
user related information in the webclient. Before this commit,
in components, they had to import the hook `useService` and in
the setup do something like `this.user = useService("user")`. In
services, they had to explicitely declare the dependency to the
user service. In any other functions (like helper functions),
it was a bit painful: the function had to take an env (or the
needed information directly) in arguments, which wasn't ideal.

This PR makes accessing user information easier. The user
service has been removed. Instead, we introduce a `user` object
which can be imported directly from everywhere. Something
similar has already been done for the rpc service. This commit
will allow to do the same for the orm service.

Part of task 3621046

### CODE REVIEW

File: addons/bus/static/src/services/bus_service.js [hunk 1]
```diff
@@ -92,7 +93,7 @@ export const busService = {
             // Let's format it in order to ease its usage:
             //     - number if user is logged, false otherwise, keep
             //       undefined to indicate session_info is not available.
-            let uid = Array.isArray(session.user_id) ? session.user_id[0] : session.user_id;
+            let uid = user.userId;
```
Comment (seb-odoo) [inline addons/bus/static/src/services/bus_service.js:96]: You might want to update the above comment if this is no longer relevant, as well as the code below.
But is the PR actually making sure the user is now consistent across all contexts?
cc @tsm-odoo

File: addons/mail/static/src/core/web/activity_list_popover.js [hunk 2]
```diff
@@ -84,7 +84,7 @@ export class ActivityListPopover extends Component {
             "activity_format",
             [props.activityIds],
             {
-                context: this.user.user_context,
+                context: user.context,
```
Comment (seb-odoo) [inline addons/mail/static/src/core/web/activity_list_popover.js:87]: Isn't/shouldn't the user context be automatically provided to the orm calls instead?

File: addons/mail/static/tests/helpers/test_utils.js [hunk 3]
```diff
@@ -3,21 +3,24 @@
...
-import { session as sessionInfo } from "@web/session";
 import { registerCleanup } from "@web/../tests/helpers/cleanup";
 import {
     clearRegistryWithCleanup,
     registryNamesToCloneWithCleanup,
     prepareRegistriesWithCleanup,
 } from "@web/../tests/helpers/mock_env";
+import {
+    patchUserWithCleanup,
+    patchUserContextWithCleanup,
```
Comment (seb-odoo) [inline addons/mail/static/tests/helpers/test_utils.js:22]: (minor) would import them alphabetically

File: addons/mail/static/tests/helpers/test_utils.js [hunk 4]
```diff
@@ -171,14 +174,13 @@ export async function start(param0 = {}) {
...
-        },
-        uid: pyEnv.currentUserId,
+    patchUserWithCleanup({
+        userId: pyEnv.currentUserId,
         name: pyEnv.currentUser?.name,
-        partner_id: pyEnv.currentPartnerId,
+        partnerId: pyEnv.currentPartnerId,
+    });
+    patchUserContextWithCleanup({
+        uid: pyEnv.currentUserId,
```
Comment (seb-odoo) [inline addons/mail/static/tests/helpers/test_utils.js:183]: This is a bit worrisome, isn't/shouldn't user context uid automatically synchronized with actual user id?

**Review by rugo-odoo — APPROVED** [review]

File: addons/mail/static/src/core/web/activity_list_popover.js (hunk 2 above)
Comment (aab-odoo) [reply addons/mail/static/src/core/web/activity_list_popover.js:87]: it is indeed, good catch :+1:

File: addons/mail/static/tests/helpers/test_utils.js (hunk 4 above)
Comment (aab-odoo) [reply addons/mail/static/tests/helpers/test_utils.js:183]: Good point. Actually, I think it, I'll remove this patch and see what happens

File: addons/bus/static/src/services/bus_service.js (hunk 1 above)
Comment (aab-odoo) [reply addons/bus/static/src/services/bus_service.js:96]: Ok actually this one shouldn't have been replaced. In session_info, in the backend, we have user_id and uid, which are basically the same. In the frontend, I see we don't have uid. I'll try to uniformize this, maybe in another PR. For now, we can keep the code as it was as my PR doesn't delete user_id from the session, so it should keep working as expected.

**Comment by sdegueldre** [comment]
@robodoo r+

**Comment by aab-odoo** [comment]
robodoo rebase-merge

**Pull request closed without merging**